### ISSUE TITLE
Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0!

### ISSUE THREAD

**Issue by LeeSnake** [KatieBuc/gnnad#6]
![image](https://github.com/KatieBuc/gnnad/assets/90833171/6eea3101-2b6f-4683-ba0f-949401d4cf19)
![image](https://github.com/KatieBuc/gnnad/assets/90833171/797ccc43-62ed-47a5-9c33-a281726c5759)
If I choose cuda as the device,I will meet this problem.Can you give me some help?

**Comment by AzizjonKasimov** [comment]
> ![image](https://private-user-images.githubusercontent.com/90833171/310080079-6eea3101-2b6f-4683-ba0f-949401d4cf19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NTE0ODEsIm5iZiI6MTcxMjY1MTE4MSwicGF0aCI6Ii85MDgzMzE3MS8zMTAwODAwNzktNmVlYTMxMDEtMmI2Zi00NjgzLWJhMGYtOTQ5NDAxZDRjZjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDA4MjYyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmYWQ2MjY3YWY2ZjNkZDczMGE2YTkwODM1ZmE2NDNhZTdiMzEyM2U2YTdkODUwYzQ5ZjQ2N2FkZDNiZDY5MDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fLCPt2OJrDrZlt7DlJrEhbZETLkg15RyKjdgvAdWbP8) ![image](https://private-user-images.githubusercontent.com/90833171/310080424-797ccc43-62ed-47a5-9c33-a281726c5759.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NTE0ODEsIm5iZiI6MTcxMjY1MTE4MSwicGF0aCI6Ii85MDgzMzE3MS8zMTAwODA0MjQtNzk3Y2NjNDMtNjJlZC00N2E1LTljMzMtYTI4MTcyNmM1NzU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDA4MjYyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzYjEwYjljNmZhNDYzMDM0YWM1YjZlZDRiNGQyZGM3ZTFjZDEzOGYyNThjNDVhODM2YzI0ZjQ5N2ZhNGEyMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Q7MUcL4tho3lBE-vuPSdQTuIlhr6H8ajXVRW6Jvmioo) If I choose cuda as the device,I will meet this problem.Can you give me some help?

Yes, the problem still exists. Authors, please address the issue
